File: Python/beadsort/gemma2-9b-it.py

import random

def beadsort(arr):
  max_val = max(arr)
  num_beads = max_val + 1
  beads = [0] * num_beads

  for num in arr:
    beads[num] += 1

  i = 0
  for j in range(num_beads):
    while beads[j] > 0:
      arr[i] = j
      beads[j] -= 1
      i += 1

  return arr

import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            arr = []
            for line in f:
                line = line.strip()
                if line.isdigit():
                    val = int(line)
                    if val >= 0:
                        arr.append(val)
    except FileNotFoundError:
        return

    beadsort(arr)
if __name__ == "__main__":
    main()

File: Java/beadsort/Gemma2_9b_it.java

import java.util.ArrayList;
import java.util.List;

public class Gemma2_9b_it {

    public static void beadSort(int[] arr) {
        int max = getMax(arr);
        int[] beads = new int[max + 1];

        for (int i = 0; i < arr.length; i++) {
            beads[arr[i]]++;
        }

        int index = 0;
        for (int i = 0; i < beads.length; i++) {
            while (beads[i] > 0) {
                arr[index++] = i;
                beads[i]--;
            }
        }
    }

    private static int getMax(int[] arr) {
        int max = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] > max) {
                max = arr[i];
            }
        }
        return max;
    }

    public static void main(String[] args) {
        List<Integer> input = new ArrayList<>();

        try (java.io.BufferedReader br = new java.io.BufferedReader(new java.io.FileReader("../../inputs.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                input.add(Integer.parseInt(line.trim()));
            }
        } catch (java.io.IOException e) {
            return;
        }

        // Convert List<Integer> to int[]
        int[] arr = input.stream().mapToInt(Integer::intValue).toArray();

        beadSort(arr);

        // Convert sorted int[] back to List<Integer> if needed
        List<Integer> sorted = new ArrayList<>();
        for (int num : arr) {
            sorted.add(num);
        }

    }
}
